pom.xml

```diff
@@ -433,356 +360,58 @@
-                  <name>integration-test</name>
-                  <value>true</value>
-                </property>
-              </properties>
-            </configuration>
-          </execution>
-        </executions>
-      </plugin>
-
-      <plugin>
-        <artifactId>maven-project-info-reports-plugin</artifactId>
-        <version>${projectinfo.plugin.version}</version>
-        <configuration>
-          <dependencyLocationsEnabled>false</dependencyLocationsEnabled>
-          <dependencyDetailsEnabled>false</dependencyDetailsEnabled>
-        </configuration>
       </plugin>
 
       <plugin>
         <groupId>com.github.github</groupId>
         <artifactId>site-maven-plugin</artifactId>
-        <version>${github.site.plugin.version}</version>
-        <configuration>
-          <message>Creating site for ${project.artifactId}, ${project.version}</message>
-          <!-- this does the trick to place every module in the correct subfolder -->
-          <path>${project.distributionManagement.site.url}</path>
-          <merge>true</merge>
-          <excludes>
-            <exclude>xref-test/**</exclude>
-            <exclude>testapidocs/**</exclude>
-          </excludes>
-        </configuration>
-        <executions>
-          <execution>
-            <id>github</id>
-            <goals>
-              <goal>site</goal>
-            </goals>
-            <phase>site-deploy</phase>
-          </execution>
-        </executions>
       </plugin>
 
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-site-plugin</artifactId>
-        <version>${maven.site.plugin.version}</version>
-        <configuration>
-          <skipDeploy>true</skipDeploy>
-        </configuration>
-        <dependencies>
-          <dependency>
-             <!--allows markdown syntax for site generation. To use it
-              place files below src/site/markdown/[filename].md -->
-            <groupId>org.apache.maven.doxia</groupId>
-            <artifactId>doxia-module-markdown</artifactId>
-            <version>${doxia.plugin.version}</version>
-          </dependency>
-        </dependencies>
       </plugin>
 
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-javadoc-plugin</artifactId>
-        <version>${javadoc.plugin.version}</version>
-        <configuration>
-          <linksource>true</linksource>
-          <quiet>true</quiet>
-        </configuration>
-        <executions>
-          <execution>
-            <id>verify-javadocs</id>
-            <goals>
-              <goal>jar</goal>
-              <goal>test-jar</goal>
-            </goals>
-          </execution>
-          <execution>
-            <id>attach-javadocs</id>
-            <goals>
-              <goal>javadoc</goal>
-            </goals>
-            <phase>site</phase>
-          </execution>
-        </executions>
       </plugin>
     </plugins>
   </build>
 
-  <profiles>
```

Comment:
I have a different understanding of how profiles work. Basically, `mvn help:effective-pom` displays only the effective profile. If you want to see the "release-sign-artifacts" profile, try this: `mvn help:effective-pom -Prelease-sign-artifacts`. The output of that includes the `maven-gpg-plugin`, which otherwise is not present.